lib/internal/errors.js

```diff
@@ -0,0 +1,258 @@
+'use strict';
+
+// The whole point behind this internal module is to allow Node.js to no
+// longer be forced to treat every error message change as a semver-major
+// change. The NodeError classes here all expose a `code` property whose
+// value identifies the error.
+
+const kCode = Symbol('code');
+
+class NodeError extends Error {
+  constructor(key, ...args) {
+    super(message(key, args));
+    this[kCode] = key;
+    Error.captureStackTrace(this, NodeError);
```

Comment:
This is to filter out the `NodeError` from the top of the stack trace so that the callsite is accurately reported. See https://nodejs.org/dist/latest-v6.x/docs/api/errors.html#errors_error_capturestacktrace_targetobject_constructoropt